wagtail/admin/viewsets/pages.py

```diff
@@ -0,0 +1,42 @@
+from django.urls import path
+
+from wagtail.admin.views.pages.listing import IndexView
+from wagtail.models import Page
+
+from .base import ViewSet
+
+
+class PageListingViewSet(ViewSet):
+    #: The view class to use for the index view; must be a subclass of ``wagtail.admin.views.pages.listing.IndexView``.
+    index_view_class = IndexView
+    model = Page
+    columns = IndexView.columns
+    filterset_class = IndexView.filterset_class
```

Comment:
e.g.

```suggestion
class PageListingViewSet(ViewSet):
    """
    A viewset to present a flat listing of all pages of a specific type.

    All attributes and methods from :class:`~wagtail.admin.viewsets.base.ViewSet`
    are available.

    For more information on how to use this class, see :ref:`custom_page_listings`.
    """
    #: The view class to use for the index view; must be a subclass of ``wagtail.admin.views.pages.listing.IndexView``.
    index_view_class = IndexView
    #: Required; the page model class that this viewset will work with.
    model = Page
    #: A list of ``wagtail.admin.ui.tables.Column`` instances for the columns in the listing.
    columns = IndexView.columns
    #: A subclass of ``wagtail.admin.filters.WagtailFilterSet``, which is a
    #: subclass of `django_filters.FilterSet <https://django-filter.readthedocs.io/en/stable/ref/filterset.html>`_.
    #: This will be passed to the ``filterset_class`` attribute of the index view.
    filterset_class = IndexView.filterset_class
```